### PR TITLE
introduce `string.as-regex` object

### PR DESCRIPTION
**Pull request by Graur**
#671

### CODE REVIEW

**Comment by Graur** [comment]
@yegor256 please, have a look

File: eo-runtime/src/main/java/EOorg/EOeolang/EOstring$EOas_regex.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ */
+@XmirObject(oname = "string.as-regex")
+public class EOstring$EOas_regex extends PhDefault {
+
+    public EOstring$EOas_regex(final Phi sigma) {
+        super(sigma);
+        this.add("φ", new AtComposite(this, rho -> {
+            final String pattern = new Param(rho).strong(String.class);
+            StringBuilder builder = new StringBuilder();
+            if (pattern.startsWith("/")) {
```
Comment (yegor256) [inline eo-runtime/src/main/java/EOorg/EOeolang/EOstring$EOas_regex.java:49]: @Graur read this: https://www.yegor256.com/2015/01/21/if-then-throw-else.html

File: eo-runtime/src/test/eo/org/eolang/string-tests.eo [hunk 2]
```diff
@@ -140,3 +141,39 @@
   eq. > @
     "hello world".as-hash
     "hello world".as-hash
+
```
Comment (yegor256) [inline eo-runtime/src/test/eo/org/eolang/string-tests.eo:144]: @Graur try this pattern too: `/[0-9]+\/\d+/`. should match `15/3` for example.

**Comment by yegor256** [comment]
@Graur a few comments above

File: eo-runtime/src/test/eo/org/eolang/string-tests.eo (hunk 2 above)
Comment (Graur) [reply eo-runtime/src/test/eo/org/eolang/string-tests.eo:144]: @yegor256 this pattern works only like this:
```
/([0-9]+/\\d+)/
.matches "15/3"
```
and my object `as-regex` works like this too:
```
"/([0-9]+/\\d+)/".as-regex
.matches "15/3"
```
About parenthesis I wrote [here](https://github.com/objectionary/eo/issues/675).

File: eo-runtime/src/main/java/EOorg/EOeolang/EOstring$EOas_regex.java (hunk 1 above)
Comment (Graur) [reply eo-runtime/src/main/java/EOorg/EOeolang/EOstring$EOas_regex.java:49]: @yegor256 fixed

**Comment by Graur** [comment]
@yegor256 I fixed the code according to your comment and add proposed test, could you please check?

**Comment by Graur** [comment]
@yegor256 could you please check again

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**